### PR TITLE
Unique location and cleanup

### LINKED ISSUE #156: [Bug]: RecoverAccounts() seems to have a race condition that throws an invalid mnemonic error
### Issue description

When we recoverAccounts() in the claiming flow, wallet.rs often throws an invalid mnemonic error. I created a script that reproduces the issue. It's not always occurring, which leads me to believe that it's a race condition. To ensure that the error occurs, run the script at least 50 times.

### Version

Latest iota-sdk version

### Expected behaviour

That the script runs consistently without problems and no error is thrown.

### Actual behaviour

After a breadth search, the depth search throws an error.

### Can the issue reliably be reproduced?

Yes

### Steps to reproduce the issue

Script that reproduces the error

[recover-account-bug.txt](https://github.com/iotaledger/iota-sdk/files/11111084/recover-account-bug.txt)

### Errors

```
Error:  {
  type: 'invalidMnemonic',
  error: 'invalid mnemonic: first account address used another seed'
}
```

### Duplicate declaration

- [X] I have searched the issues tracker this issue and there is none

### PR DESCRIPTION
**Pull request by Thoralf-M**
# Description of change

Unique location and cleanup
Also fix offline signing examples

## Links to any relevant issues

Fixes https://github.com/iotaledger/iota-sdk/issues/156

## Type of change

- Bug fix (a non-breaking change which fixes an issue)

## How the change has been tested

Calling recover_accounts(), worked where it didn't before

## Change checklist

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by thibault-martinez — CHANGES_REQUESTED** [review]
CI fails

File: sdk/src/client/stronghold/secret.rs [hunk 1]
```diff
@@ -49,13 +49,22 @@ impl SecretManage for StrongholdAdapter {
...
         for address_index in address_indexes {
-            let chain = Chain::from_u32_hardened(vec![44u32, coin_type, account_index, internal as u32, address_index]);
+            let bip_path = vec![44u32, coin_type, account_index, internal as u32, address_index];
+            let chain = Chain::from_u32_hardened(bip_path);
+
+            let derive_location = Location::generic(
+                SECRET_VAULT_PATH,
+                [
+                    DERIVE_OUTPUT_RECORD_PATH,
+                    &chain.segments().iter().flat_map(|seg| seg.bs()).collect::<Vec<u8>>(),
```
Comment (thibault-martinez) [inline sdk/src/client/stronghold/secret.rs:65]: Do you think this is a common operation that we should put in crypto.rs ?

Comment (Thoralf-M) [reply sdk/src/client/stronghold/secret.rs:65]: No, don't see other use cases that would do this

**Review by thibault-martinez — DISMISSED** [review]

**Review by thibault-martinez — DISMISSED** [review]

File: sdk/examples/wallet/offline_signing/0_generate_addresses.rs [hunk 2]
```diff
@@ -31,7 +31,9 @@ async fn main() -> Result<()> {
     // Setup Stronghold secret_manager
     let mut secret_manager = StrongholdSecretManager::builder()
         .password(&std::env::var("STRONGHOLD_PASSWORD").unwrap())
-        .build(PathBuf::from("examples/offline_signing/offline_signing.stronghold"))?;
+        .build(PathBuf::from(
+            "examples/wallet/offline_signing/offline_signing.stronghold",
+        ))?;
```
Comment (Alex6323) [inline sdk/examples/wallet/offline_signing/0_generate_addresses.rs]: `Path::new("examples/...")`

File: sdk/examples/wallet/offline_signing/2_sign_transaction.rs [hunk 3]
```diff
@@ -34,7 +34,9 @@ async fn main() -> Result<()> {
     // Setup Stronghold secret_manager
     let mut secret_manager = StrongholdSecretManager::builder()
         .password(&std::env::var("STRONGHOLD_PASSWORD").unwrap())
-        .build(PathBuf::from("examples/offline_signing/offline_signing.stronghold"))?;
+        .build(PathBuf::from(
+            "examples/wallet/offline_signing/offline_signing.stronghold",
+        ))?;
```
Comment (Alex6323) [inline sdk/examples/wallet/offline_signing/2_sign_transaction.rs]: same

File: sdk/examples/wallet/01_create_wallet.rs [hunk 4]
```diff
@@ -24,7 +24,7 @@ async fn main() -> Result<()> {
     // Setup Stronghold secret_manager
     let mut secret_manager = StrongholdSecretManager::builder()
         .password(&std::env::var("STRONGHOLD_PASSWORD").unwrap())
-        .build(PathBuf::from("wallet.stronghold"))?;
+        .build(Path::new("wallet.stronghold"))?;
```
Comment (thibault-martinez) [inline sdk/examples/wallet/01_create_wallet.rs]: `build` takes a `AsRef<Path>` anyway so you don't even need a `Path`, just a `&str` should work

Comment (thibault-martinez) [reply sdk/examples/wallet/01_create_wallet.rs]: Can you do a search on the whole repo if we do that elsewhere pls

Comment (thibault-martinez) [reply sdk/examples/wallet/01_create_wallet.rs]: Would be nice if Clippy could see that

Comment (Thoralf-M) [reply sdk/examples/wallet/01_create_wallet.rs]: Updated

**Review by thibault-martinez — APPROVED** [review]

**Review by Alex6323 — APPROVED** [review]

**Review by kwek20 — APPROVED** [review]

**Pull request merged**